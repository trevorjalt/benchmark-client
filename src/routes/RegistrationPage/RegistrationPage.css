.RegistrationTitle {
    font-size: 3.5vh;
}