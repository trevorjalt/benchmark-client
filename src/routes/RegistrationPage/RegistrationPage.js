import React, { Component } from 'react'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'
import './RegistrationPage.css'

export default class RegistrationPage extends Component {
    static defaultProps = {
        history: {
            push: () => {},
        },
    }

    handleRegistrationSuccess = user => {
        const { history } = this.props
        history.push('/login')
    }

    render() {
        return (
            <section className='RegistrationPage'>
                <h1 className='RegistrationTitle'>Register</h1>
                <RegistrationForm
                    onRegistrationSuccess={this.handleRegistrationSuccess}
                />
            </section>
        )
    }
}