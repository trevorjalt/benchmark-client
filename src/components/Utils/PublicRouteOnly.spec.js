import React from 'react'
import PublicOnlyRoute from './PublicOnlyRoute'
import RegistrationPage from '../../Routes/RegistrationPage/RegistrationPage'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`PublicOnlyRoute Component`, () => {
    const props = {
        component: <RegistrationPage />
    }
    
    it (`renders a public only route`, () => {
        const wrapper = 
            shallow (
                <PublicOnlyRoute {...props} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})