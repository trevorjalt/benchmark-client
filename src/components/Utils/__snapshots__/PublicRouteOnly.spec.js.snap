// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicOnlyRoute Component renders a public only route 1`] = `
<Route
  render={[Function]}
/>
`;
